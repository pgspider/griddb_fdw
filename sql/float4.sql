--
-- FLOAT4
--
CREATE EXTENSION griddb_fdw;
CREATE SERVER griddb_svr FOREIGN DATA WRAPPER griddb_fdw OPTIONS(host '239.0.0.1', port '31999', clustername 'griddbfdwTestCluster');
CREATE USER MAPPING FOR public SERVER griddb_svr OPTIONS(username 'admin', password 'testadmin');
CREATE FOREIGN TABLE FLOAT4_TBL(id serial OPTIONS (rowkey 'true'), f1 float4) SERVER griddb_svr;

INSERT INTO FLOAT4_TBL(f1) VALUES ('    0.0');
INSERT INTO FLOAT4_TBL(f1) VALUES ('1004.30   ');
INSERT INTO FLOAT4_TBL(f1) VALUES ('     -34.84    ');
INSERT INTO FLOAT4_TBL(f1) VALUES ('1.2345678901234e+20');
INSERT INTO FLOAT4_TBL(f1) VALUES ('1.2345678901234e-20');

-- test for over and under flow
INSERT INTO FLOAT4_TBL(f1) VALUES ('10e70');
INSERT INTO FLOAT4_TBL(f1) VALUES ('-10e70');
INSERT INTO FLOAT4_TBL(f1) VALUES ('10e-70');
INSERT INTO FLOAT4_TBL(f1) VALUES ('-10e-70');

-- bad input
INSERT INTO FLOAT4_TBL(f1) VALUES ('');
INSERT INTO FLOAT4_TBL(f1) VALUES ('       ');
INSERT INTO FLOAT4_TBL(f1) VALUES ('xyz');
INSERT INTO FLOAT4_TBL(f1) VALUES ('5.0.0');
INSERT INTO FLOAT4_TBL(f1) VALUES ('5 . 0');
INSERT INTO FLOAT4_TBL(f1) VALUES ('5.   0');
INSERT INTO FLOAT4_TBL(f1) VALUES ('     - 3.0');
INSERT INTO FLOAT4_TBL(f1) VALUES ('123            5');

-- special inputs
BEGIN;
DELETE FROM FLOAT4_TBL;
INSERT INTO FLOAT4_TBL(f1) VALUES ('NaN'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('nan'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('   NAN  '::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('infinity'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('          -INFINiTY   '::float4);
SELECT f1 AS float4 FROM FLOAT4_TBL;
ROLLBACK;
-- bad special inputs
INSERT INTO FLOAT4_TBL(f1) VALUES ('N A N'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('NaN x'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES (' INFINITY    x'::float4);

BEGIN;
DELETE FROM FLOAT4_TBL;
INSERT INTO FLOAT4_TBL(f1) VALUES ('Infinity'::float4 + 100.0);
INSERT INTO FLOAT4_TBL(f1) VALUES ('Infinity'::float4 / 'Infinity'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('nan'::float4 / 'nan'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('nan'::numeric::float4);
SELECT f1 AS float4 FROM FLOAT4_TBL;
ROLLBACK;

SELECT '' AS five, * FROM FLOAT4_TBL;

SELECT '' AS four, f.* FROM FLOAT4_TBL f WHERE f.f1 <> '1004.3';

SELECT '' AS one, f.* FROM FLOAT4_TBL f WHERE f.f1 = '1004.3';

SELECT '' AS three, f.* FROM FLOAT4_TBL f WHERE '1004.3' > f.f1;

SELECT '' AS three, f.* FROM FLOAT4_TBL f WHERE  f.f1 < '1004.3';

SELECT '' AS four, f.* FROM FLOAT4_TBL f WHERE '1004.3' >= f.f1;

SELECT '' AS four, f.* FROM FLOAT4_TBL f WHERE  f.f1 <= '1004.3';

SELECT '' AS three, f.f1, f.f1 * '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';

SELECT '' AS three, f.f1, f.f1 + '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';

SELECT '' AS three, f.f1, f.f1 / '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';

SELECT '' AS three, f.f1, f.f1 - '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';

-- test divide by zero
SELECT '' AS bad, f.f1 / '0.0' from FLOAT4_TBL f;

SELECT '' AS five, * FROM FLOAT4_TBL;

-- test the unary float4abs operator
SELECT '' AS five, f.f1, @f.f1 AS abs_f1 FROM FLOAT4_TBL f;

UPDATE FLOAT4_TBL
   SET f1 = FLOAT4_TBL.f1 * '-1'
   WHERE FLOAT4_TBL.f1 > '0.0';

SELECT '' AS five, * FROM FLOAT4_TBL;

DROP FOREIGN TABLE FLOAT4_TBL;
DROP USER MAPPING FOR public SERVER griddb_svr;
DROP SERVER griddb_svr;
DROP EXTENSION griddb_fdw CASCADE;