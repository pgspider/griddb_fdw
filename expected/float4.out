--
-- FLOAT4
--
CREATE EXTENSION griddb_fdw;
CREATE SERVER griddb_svr FOREIGN DATA WRAPPER griddb_fdw OPTIONS(host '239.0.0.1', port '31999', clustername 'griddbfdwTestCluster');
CREATE USER MAPPING FOR public SERVER griddb_svr OPTIONS(username 'admin', password 'testadmin');
CREATE FOREIGN TABLE FLOAT4_TBL(id serial OPTIONS (rowkey 'true'), f1 float4) SERVER griddb_svr;
INSERT INTO FLOAT4_TBL(f1) VALUES ('    0.0');
INSERT INTO FLOAT4_TBL(f1) VALUES ('1004.30   ');
INSERT INTO FLOAT4_TBL(f1) VALUES ('     -34.84    ');
INSERT INTO FLOAT4_TBL(f1) VALUES ('1.2345678901234e+20');
INSERT INTO FLOAT4_TBL(f1) VALUES ('1.2345678901234e-20');
-- test for over and under flow
INSERT INTO FLOAT4_TBL(f1) VALUES ('10e70');
ERROR:  value out of range: overflow
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('10e70');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('-10e70');
ERROR:  value out of range: overflow
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('-10e70');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('10e-70');
ERROR:  value out of range: underflow
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('10e-70');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('-10e-70');
ERROR:  value out of range: underflow
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('-10e-70');
                                           ^
-- bad input
INSERT INTO FLOAT4_TBL(f1) VALUES ('');
ERROR:  invalid input syntax for type real: ""
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('       ');
ERROR:  invalid input syntax for type real: "       "
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('       ');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('xyz');
ERROR:  invalid input syntax for type real: "xyz"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('xyz');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('5.0.0');
ERROR:  invalid input syntax for type real: "5.0.0"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('5.0.0');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('5 . 0');
ERROR:  invalid input syntax for type real: "5 . 0"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('5 . 0');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('5.   0');
ERROR:  invalid input syntax for type real: "5.   0"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('5.   0');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('     - 3.0');
ERROR:  invalid input syntax for type real: "     - 3.0"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('     - 3.0');
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('123            5');
ERROR:  invalid input syntax for type real: "123            5"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('123            5');
                                           ^
-- special inputs
BEGIN;
DELETE FROM FLOAT4_TBL;
INSERT INTO FLOAT4_TBL(f1) VALUES ('NaN'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('nan'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('   NAN  '::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('infinity'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('          -INFINiTY   '::float4);
SELECT f1 AS float4 FROM FLOAT4_TBL;
  float4   
-----------
       NaN
       NaN
       NaN
  Infinity
 -Infinity
(5 rows)

ROLLBACK;
-- bad special inputs
INSERT INTO FLOAT4_TBL(f1) VALUES ('N A N'::float4);
ERROR:  invalid input syntax for type real: "N A N"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('N A N'::float4);
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES ('NaN x'::float4);
ERROR:  invalid input syntax for type real: "NaN x"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES ('NaN x'::float4);
                                           ^
INSERT INTO FLOAT4_TBL(f1) VALUES (' INFINITY    x'::float4);
ERROR:  invalid input syntax for type real: " INFINITY    x"
LINE 1: INSERT INTO FLOAT4_TBL(f1) VALUES (' INFINITY    x'::float4)...
                                           ^
BEGIN;
DELETE FROM FLOAT4_TBL;
INSERT INTO FLOAT4_TBL(f1) VALUES ('Infinity'::float4 + 100.0);
INSERT INTO FLOAT4_TBL(f1) VALUES ('Infinity'::float4 / 'Infinity'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('nan'::float4 / 'nan'::float4);
INSERT INTO FLOAT4_TBL(f1) VALUES ('nan'::numeric::float4);
SELECT f1 AS float4 FROM FLOAT4_TBL;
  float4  
----------
 Infinity
      NaN
      NaN
      NaN
(4 rows)

ROLLBACK;
SELECT '' AS five, * FROM FLOAT4_TBL;
 five | id |     f1      
------+----+-------------
      |  1 |           0
      |  2 |      1004.3
      |  3 |      -34.84
      |  4 | 1.23457e+20
      |  5 | 1.23457e-20
(5 rows)

SELECT '' AS four, f.* FROM FLOAT4_TBL f WHERE f.f1 <> '1004.3';
 four | id |     f1      
------+----+-------------
      |  1 |           0
      |  2 |      1004.3
      |  3 |      -34.84
      |  4 | 1.23457e+20
      |  5 | 1.23457e-20
(5 rows)

SELECT '' AS one, f.* FROM FLOAT4_TBL f WHERE f.f1 = '1004.3';
 one | id | f1 
-----+----+----
(0 rows)

SELECT '' AS three, f.* FROM FLOAT4_TBL f WHERE '1004.3' > f.f1;
 three | id |     f1      
-------+----+-------------
       |  1 |           0
       |  2 |      1004.3
       |  3 |      -34.84
       |  5 | 1.23457e-20
(4 rows)

SELECT '' AS three, f.* FROM FLOAT4_TBL f WHERE  f.f1 < '1004.3';
 three | id |     f1      
-------+----+-------------
       |  1 |           0
       |  2 |      1004.3
       |  3 |      -34.84
       |  5 | 1.23457e-20
(4 rows)

SELECT '' AS four, f.* FROM FLOAT4_TBL f WHERE '1004.3' >= f.f1;
 four | id |     f1      
------+----+-------------
      |  1 |           0
      |  2 |      1004.3
      |  3 |      -34.84
      |  5 | 1.23457e-20
(4 rows)

SELECT '' AS four, f.* FROM FLOAT4_TBL f WHERE  f.f1 <= '1004.3';
 four | id |     f1      
------+----+-------------
      |  1 |           0
      |  2 |      1004.3
      |  3 |      -34.84
      |  5 | 1.23457e-20
(4 rows)

SELECT '' AS three, f.f1, f.f1 * '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';
 three |     f1      |      x       
-------+-------------+--------------
       |      1004.3 |       -10043
       | 1.23457e+20 | -1.23457e+21
       | 1.23457e-20 | -1.23457e-19
(3 rows)

SELECT '' AS three, f.f1, f.f1 + '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';
 three |     f1      |      x      
-------+-------------+-------------
       |      1004.3 |       994.3
       | 1.23457e+20 | 1.23457e+20
       | 1.23457e-20 |         -10
(3 rows)

SELECT '' AS three, f.f1, f.f1 / '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';
 three |     f1      |      x       
-------+-------------+--------------
       |      1004.3 |      -100.43
       | 1.23457e+20 | -1.23457e+19
       | 1.23457e-20 | -1.23457e-21
(3 rows)

SELECT '' AS three, f.f1, f.f1 - '-10' AS x FROM FLOAT4_TBL f
   WHERE f.f1 > '0.0';
 three |     f1      |      x      
-------+-------------+-------------
       |      1004.3 |      1014.3
       | 1.23457e+20 | 1.23457e+20
       | 1.23457e-20 |          10
(3 rows)

-- test divide by zero
SELECT '' AS bad, f.f1 / '0.0' from FLOAT4_TBL f;
ERROR:  division by zero
SELECT '' AS five, * FROM FLOAT4_TBL;
 five | id |     f1      
------+----+-------------
      |  1 |           0
      |  2 |      1004.3
      |  3 |      -34.84
      |  4 | 1.23457e+20
      |  5 | 1.23457e-20
(5 rows)

-- test the unary float4abs operator
SELECT '' AS five, f.f1, @f.f1 AS abs_f1 FROM FLOAT4_TBL f;
 five |     f1      |   abs_f1    
------+-------------+-------------
      |           0 |           0
      |      1004.3 |      1004.3
      |      -34.84 |       34.84
      | 1.23457e+20 | 1.23457e+20
      | 1.23457e-20 | 1.23457e-20
(5 rows)

UPDATE FLOAT4_TBL
   SET f1 = FLOAT4_TBL.f1 * '-1'
   WHERE FLOAT4_TBL.f1 > '0.0';
SELECT '' AS five, * FROM FLOAT4_TBL;
 five | id |      f1      
------+----+--------------
      |  1 |            0
      |  2 |      -1004.3
      |  3 |       -34.84
      |  4 | -1.23457e+20
      |  5 | -1.23457e-20
(5 rows)

DROP FOREIGN TABLE FLOAT4_TBL;
DROP USER MAPPING FOR public SERVER griddb_svr;
DROP SERVER griddb_svr;
DROP EXTENSION griddb_fdw CASCADE;
